### ISSUE TITLE
Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined

### ISSUE THREAD

**Issue by nsavelyeva** [megalinter/megalinter#712]
**Describe the bug**
Using GitHub Actions, Groovy linter fails with the below error on every Groovy file:
`Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined`.
Locally this error is not reproduced.

**To Reproduce**

1. Have a maven project with some groovy files.
2. Create a GitHub Action workflow, in my case I configured the job as follows:
```
jobs:
  cancel_duplicates:
    name: Cancel duplicate jobs
    runs-on: ubuntu-latest
    steps:
      - uses: fkirc/skip-duplicate-actions@master
        with:
          github_token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}

  build:
    name: Mega-Linter src/main
    runs-on: ubuntu-latest
    steps:
      - name: Checkout Code
        uses: actions/checkout@v2
        with:
          token: ${{ secrets.PAT || secrets.GITHUB_TOKEN }}
          fetch-depth: 0

      - name: Mega-Linter
        id: ml
        uses: nvuillam/mega-linter@v4
        env:
          ENABLE: GROOVY
          GROOVY_NPM_GROOVY_LINT_ARGUMENTS: '--output report/megalinter-src-main.html'
          GROOVY_NPM_GROOVY_LINT_CONFIG_FILE: .groovylintrc-src-main.json
          GROOVY_NPM_GROOVY_LINT_RULES_PATH: conf
          GROOVY_NPM_GROOVY_LINT_FILTER_REGEX_INCLUDE: src/main
          PRINT_ALPACA: false
          VALIDATE_ALL_CODEBASE: true
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
3. Trigger the GitHub Action and wait until it gets completed.
4. Open checks tab as shown in the screenshot below.
5. See the `totalFoundErrorNumber` error on every file

**Expected behavior**
GitHub Actions produce no `totalFoundErrorNumber` errors, and has the same output as the local execution.

**Screenshots**
![Screenshot 2021-09-01 at 11 12 02](https://user-images.githubusercontent.com/21998320/131645592-f12f0433-7a12-4d30-950d-4c293c4808d2.png)

**Additional context**
- For local execution I use the same parameters:
```
npx mega-linter-runner -r insiders -e 'ENABLE=GROOVY' -e 'GROOVY_NPM_GROOVY_LINT_FILTER_REGEX_INCLUDE=src/main' -e 'GROOVY_NPM_GROOVY_LINT_CONFIG_FILE=conf/.groovylintrc-src-main.json' -e 'GROOVY_NPM_GROOVY_LINT_RULES_PATH=conf' -e 'PRINT_ALPACA=false' -e 'SHOW_ELAPSED_TIME=true'
``` 
- I found a similar error in [this closed issue](https://github.com/nvuillam/npm-groovy-lint/issues/107).

**Comment by nvuillam** [comment]
That's a npm-groovy-lint bug :/

I'l also the maintainer of npm-groovy-lint (but i have less available time), i'll have a look !

**Comment by nsavelyeva** [comment]
An update from my side.
Apparently, the error is caused by `GROOVY_NPM_GROOVY_LINT_ARGUMENTS` option.
If I remove it, then GitHub Action workflows work as expected.
In my initial description of the issue the local execution command does not have that option and that is why the error is not observed there.
But if the option is used (as shown in the command below), then local execution will also be affected with this error.
```
npx mega-linter-runner -r insiders -e 'ENABLE=GROOVY' -e 'GROOVY_NPM_GROOVY_LINT_FILTER_REGEX_INCLUDE=src/main' -e 'GROOVY_NPM_GROOVY_LINT_CONFIG_FILE=conf/.groovylintrc-src-main.json' -e 'GROOVY_NPM_GROOVY_LINT_RULES_PATH=conf' -e 'GROOVY_NPM_GROOVY_LINT_ARGUMENTS=--output "report/megalinter-src-main.html"' -e 'PRINT_ALPACA=false' -e 'SHOW_ELAPSED_TIME=true'
```

**Comment by GordonJess** [comment]
I'm also having this issue after enabling HTML report output with npm-groovy-lint!

**Comment by nvuillam** [comment]
@nsavelyeva @GordonJess is it now better with npm-groovy-lint v9.0.0 ?

**Comment by nsavelyeva** [comment]
Hi @nvuillam, thanks for the response.
I executed this command:
```
npx mega-linter-runner -r insiders -e 'ENABLE=GROOVY' -e 'GROOVY_NPM_GROOVY_LINT_FILTER_REGEX_EXCLUDE=Jenkinsfile.*' -e 'PRINT_ALPACA=false' -e 'SHOW_ELAPSED_TIME=true' -e 'GROOVY_NPM_GROOVY_LINT_ARGUMENTS=--output report/megalinter-src-main.html'
```
In the output I see it is using v9.0.0: `Using [npm-groovy-lint v9.0.0] https://nvuillam.github.io/npm-groovy-lint/`
But the error is still reproduced:
```
GroovyLint: Started CodeNarc Server
GroovyLint: Successfully processed CodeNarc: 
CodeNarc completed: (p1=0; p2=2; p3=11) 4597ms

Unexpected error: Cannot read property 'totalFoundErrorNumber' of undefined
TypeError: Cannot read property 'totalFoundErrorNumber' of undefined
    at NpmGroovyLint.manageReturnCode (/node_modules/npm-groovy-lint/lib/groovy-lint.js:479:49)
    at NpmGroovyLint.postProcess (/node_modules/npm-groovy-lint/lib/groovy-lint.js:328:14)
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
    at async NpmGroovyLint.run (/node_modules/npm-groovy-lint/lib/groovy-lint.js:66:13)
    at async /node_modules/npm-groovy-lint/lib/index.js:12:9
```

**Comment by nvuillam** [comment]
mmmm I think that the solution must be done within npm-groovy-lint instead of Mega-Linter
